File: frontend/src/types/evidenceUnit.ts
/** Mirrors `EvidenceUnitResponse` / `EvidenceUnitListResponse` from `app/schemas/evidence_unit.py`. */

export interface EvidenceUnitResponse {
  id: string;
  artifact_id: string;
  modality: string;
  content_type: string;
  text: string | null;
  location: Record<string, unknown> | null;
  source_fidelity: string;
  confidence: number | null;
  created_at: string;
  updated_at: string;
}

export interface EvidenceUnitListResponse {
  items: EvidenceUnitResponse[];
}

File: backend/app/schemas/evidence_unit.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class EvidenceUnitResponse(BaseModel):
    """EvidenceUnit response (Phase 2 inspection)."""

    id: str
    artifact_id: str
    modality: str
    content_type: str
    text: str | None
    location: dict[str, Any] | None
    source_fidelity: str
    confidence: float | None
    created_at: datetime
    updated_at: datetime


class EvidenceUnitListResponse(BaseModel):
    """List evidence units response."""

    items: list[EvidenceUnitResponse]

